### PR TITLE
fix(pricing_rules): pricing rule selector is wrong

### PR DESCRIPTION
**Pull request by pipech**
Pricing rule with different `apply_on` won't work properly.

Example:
- Rule A: Discount 10% on All Product Group [Priority 1]

image

- Rule B: Discount 20% on Brand A [Priority 20]

image

These rules won't work properly because `get_pricing_rule` function will stop getting rule as soon as it found matching rule but there are a case like these one that need to get rule across ['item_code', 'item_group', 'brand'].

So what we do here is we get all rules across ['item_code', 'item_group', 'brand'] then remove duplicate rules if there are one.

### CODE REVIEW

**Comment by rohitwaghchaure** [comment]
@pipech travis is failing, kindly fix the failed test case

File: erpnext/accounts/doctype/pricing_rule/utils.py [hunk 1]
```diff
@@ -30,13 +30,19 @@ class MultiplePricingRuleConflict(frappe.ValidationError): pass
...
-		pricing_rules.extend(_get_pricing_rules(apply_on, args, values))
-		if pricing_rules and not apply_multiple_pricing_rules(pricing_rules):
-			break
+		pricing_rules_all.extend(_get_pricing_rules(apply_on, args, values))
+
+	# removing duplicate pricing rule
+	pricing_rules_title = []
+	pricing_rules = []
+	for p in pricing_rules_all:
+		if p['title'] not in pricing_rules_title:
```
Comment (rohitwaghchaure) [inline erpnext/accounts/doctype/pricing_rule/utils.py]: use "name"(primary key) instead of "title"

Comment (pipech) [reply erpnext/accounts/doctype/pricing_rule/utils.py]: I've change `title` to `name`

https://github.com/frappe/erpnext/pull/22328/commits/303639db764672ba9347475419c347aad6d06e02

**Comment by marination** [comment]
@pipech Please resolve conflicts

**Comment by barredterra** [comment]
@pipech still want to get this merged?

@rohitwaghchaure can we remove the don't merge label?

**Review by rohitwaghchaure — APPROVED** [review]

**Pull request merged**